tests/test1_login_test.py

```diff
@@ -0,0 +1,139 @@
+        
+    def configure_test_2_4_LoginFailure_by_incorrect_password():
+        (username, password) = ("sampleuser_Login@mail.com","a"*8)
+        return (username, password)
+    
+
+class TestLogin(unittest.TestCase):
+    
+    
+    
+    def test_1_SignUp_Existed(self):
+        #This test would work for both first time creating a new user and duplicate creation. The duplicate running would assert the error message.
+        
+        #only print out when error happens
+        print("\n\nTesting SignUp\n\n") 
+        
+        testSignUp = signUp()
+        (username, password) = configure.configure_test_2_0_successOrExisted()
+        (urlCurrent, alertInfo) = testSignUp.Driver_SignUp(username, password)        
+        testSignUp.Close()
+        
+
+    
+    def test_signUp_loginLink(self):
+        
+        logInPage = logIn() 
+        signUpUrl = logInPage.Driver_Login_signUpLink()
+        
+        isLoginUrl = signUpUrl == "http://localhost:5000/signup"
+        
+        self.assertTrue(isLoginUrl)
+    
+    
+    
+    
+    
+    
+    def test_2_0_LoginSuccess(self):
+        #successfully login - with correct username and password
+        print("\n\nTesting LoginSuccess\n\n")
+        
+        logInPage = logIn()     # constuctor
+        (username, password) = configure.configure_test_2_0_successOrExisted()
+        currentUrl = logInPage.LoginAttempt(username, password)
+        logInPage.Close()
+        
+        
+        IsLoginSuccess = (currentUrl == "http://localhost:5000/instructor_project")
+        
+        self.assertTrue(IsLoginSuccess)   
+       
+    def test_2_1_LoginFailure_by_user_not_exist(self):
+        #failed login due to "user doesn't exist"
+
+        print("\n\nTesting LoginFailure1 - due to 'user doesn't exist'\n\n")
+        
+        logInPage = logIn()
+        (username, password) = configure.configure_test_2_1_LoginFailure_by_user_not_exist()
+        currentUrl = logInPage.LoginAttempt(username, password)
+        alertInfo = logInPage.getUserExistAlert()  # no need for self -- passed already through logInPage
+        logInPage.Close()
+        
+        IsUrlTrue = currentUrl == "http://localhost:5000/login"
+        IsAlert = alertInfo == "user doesn't exist"
+        
+        self.assertTrue(IsUrlTrue and IsAlert)
+        
+   
+    def test_2_2_LoginFailure_by_invalid_password(self):
+        #failed login due to password too short or too long(should be between 8 - 80)
+
+        print("\n\nTesting LoginFailure1 - due to 'user doesn't exist'\n\n")
+        
+        logInPage = logIn()
+        (username, password) = configure.configure_test_2_2_LoginFailure_by_invalid_password()
+        currentUrl = logInPage.LoginAttempt(username, password)
+        alertInfo = logInPage.getPasswordAlert()  # no need for self -- passed already through logInPage
+        logInPage.Close()
+        
+        IsUrlTrue = currentUrl == "http://localhost:5000/login"
```

Comment:
refactoring all variable names

should use same convention as function names, that is, lowercase with underscores (e.g., alert_info)

Reply:
use IDEs refactoring tools to do this quickly

Reply:
Current python styling conventions are now followed throughout the test suite.